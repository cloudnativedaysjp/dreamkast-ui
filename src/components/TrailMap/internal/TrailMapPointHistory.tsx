import React from 'react'
import * as Styled from '../styled'
import { pointsSelector } from '../../../store/points'
import { useSelector } from 'react-redux'
import {
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
} from '@material-ui/core'

export const TrailMapPointHistory = () => {
  const { pointData } = useSelector(pointsSelector)
  return (
    <Styled.PointHistoryTable>
      <TableContainer component={Paper}>
        <Table aria-label="event-history" size="small">
          <TableHead>
            <TableRow>
              <TableCell>獲得ポイント</TableCell>
              <TableCell>達成条件</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {pointData.points.map((p) => {
              return (
                <TableRow key={p.pointEventId}>
                  <TableCell component="th" scope="row">
                    {p.point}pt
                  </TableCell>
                  <TableCell>{p.desc}</TableCell>
                </TableRow>
              )
            })}
          </TableBody>
        </Table>
      </TableContainer>
    </Styled.PointHistoryTable>
  )
}
