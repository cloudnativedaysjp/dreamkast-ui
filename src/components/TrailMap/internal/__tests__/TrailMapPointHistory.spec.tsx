import React from 'react'
import { render } from '@testing-library/react'
import { TrailMapPointHistory } from '../TrailMapPointHistory'

describe('TrailMapPointHistory', () => {
  it('should render without crash', () => {
    const data = {
      total: 70,
      points: [
        {
          conference: 'cicd2023',
          pointEventId: '34973274ccef6ab4dfaaf86599792fa9c3fe4689',
          point: 50,
          timestamp: 1678197315,
          desc: 'セッション視聴: 1日目 スロット1',
        },
        {
          conference: 'cicd2023',
          pointEventId: '23e182506f4b883d8aae3d29d08e044c55b04deb',
          point: 20,
          timestamp: 1678197330,
          desc: 'セッション視聴: 1日目 スロット2',
        },
      ],
    }
    const screen = render(
      <TrailMapPointHistory pointData={data}></TrailMapPointHistory>,
    )
    expect(screen.asFragment()).toMatchSnapshot()
  })
})
