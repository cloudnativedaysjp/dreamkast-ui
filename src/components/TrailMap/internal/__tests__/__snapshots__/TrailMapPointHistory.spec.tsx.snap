// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrailMapPointHistory should render without crash 1`] = `
<DocumentFragment>
  <div
    class="styled__PointHistoryTable-sc-1t1auio-12 gDdnXT"
  >
    <div
      class="MuiTableContainer-root"
    >
      <table
        aria-label="event-history"
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-1 MuiTableCell-sizeSmall"
              scope="col"
            >
              獲得ポイント
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-1 MuiTableCell-sizeSmall"
              scope="col"
            >
              達成条件
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2 MuiTableCell-sizeSmall"
              role="cell"
              scope="row"
            >
              50pt
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2 MuiTableCell-sizeSmall"
            >
              セッション視聴: 1日目 スロット1
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2 MuiTableCell-sizeSmall"
              role="cell"
              scope="row"
            >
              20pt
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2 MuiTableCell-sizeSmall"
            >
              セッション視聴: 1日目 スロット2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
